### PR TITLE
Backport ICM20602 Fast Reset and make it optional

### PR DESCRIPTION
**Pull request by bugobliterator**
Enabled only on CubeOrange and related boards

Also the ICM20602 and ICM20948_ext have been swapped in order.

### CODE REVIEW

File: libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp [hunk 1]
```diff
@@ -46,6 +47,10 @@ extern const AP_HAL::HAL& hal;
 #define debug(fmt, args ...)  do {printf("MPU: " fmt "\n", ## args); } while(0)
 #endif
 
+#ifndef HAL_ENABLE_FAST_FIFO_RESET_ICM20602
+#define HAL_ENABLE_FAST_FIFO_RESET_ICM20602 false
```
Comment (andyp1per) [inline libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp:51]: I think this should be 1/0 or TRUE/FALSE - I don't think we use lower case anywhere

Comment (andyp1per) [reply libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp:51]: Not seeing where this is actually used?

File: libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp [hunk 2]
```diff
@@ -195,6 +200,16 @@ void AP_InertialSensor_Invensense::_fifo_reset(bool log_error)
     notify_gyro_fifo_reset(_gyro_instance);
 }
 
+void AP_InertialSensor_Invensense::_fast_fifo_reset()
```
Comment (andyp1per) [inline libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp:203]: #if out with HAL_ENABLE_FAST_FIFO_RESET_ICM20602

File: libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp [hunk 3]
```diff
@@ -434,6 +453,28 @@ bool AP_InertialSensor_Invensense::update() /* front end */
 
     _publish_temperature(_accel_instance, _temp_filtered);
 
+    if (fast_reset_count) {
```
Comment (andyp1per) [inline libraries/AP_InertialSensor/AP_InertialSensor_Invensense.cpp:456]: #if out with HAL_ENABLE_FAST_FIFO_RESET_ICM20602

**Review by andyp1per — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef.inc [hunk 4]
```diff
@@ -243,8 +243,8 @@ IMU LSM9DS0 SPI:lsm9ds0_ext_g SPI:lsm9ds0_ext_am ROTATION_ROLL_180_YAW_270 ROTAT
 IMU Invensense SPI:mpu9250 ROTATION_YAW_270
 
 # alternative IMU set for newer cubes
-IMU Invensense SPI:icm20602_ext ROTATION_ROLL_180_YAW_270
 IMU Invensensev2 SPI:icm20948_ext ROTATION_PITCH_180
+IMU Invensense SPI:icm20602_ext ROTATION_ROLL_180_YAW_270
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef.inc:247]: need to use EK3_PRIMARY and EK2_PRIMARY

File: libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef.inc [hunk 5]
```diff
@@ -340,3 +340,5 @@ define HAL_ADSB_SAGETECH_MXS_ENABLED HAL_ADSB_ENABLED
 # note that if firmware is build with --secure-bl then DFU is
 # disabled
 ENABLE_DFU_BOOT 1
+
+define HAL_ENABLE_FAST_FIFO_RESET_ICM20602 true
```
Comment (tridge) [inline libraries/AP_HAL_ChibiOS/hwdef/CubeOrange/hwdef.inc:344]: prefer 1 and 0 for true and false

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by tridge** [comment]
@bugobliterator we're about to do copter 4.3.1beta1 - can you update this PR to be ready for backport?

**Comment by bugobliterator** [comment]
> @bugobliterator we're about to do copter 4.3.1beta1 - can you update this PR to be ready for backport?

@tridge done.

**Comment by rmackay9** [comment]
@tridge @bugobliterator,

Shall we merge this one to the 4.3 branch now?  I'm thinking about pushing out 4.3.3-rc1 in the coming days.

**Comment by bugobliterator** [comment]
> @tridge @bugobliterator,
> 
> Shall we merge this one to the 4.3 branch now?  I'm thinking about pushing out 4.3.3-rc1 in the coming days.

@rmackay9 I am fine with this coming into next rc.

**Comment by rmackay9** [comment]
This is included in 4.3.3-rc1

**Comment by rmackay9** [comment]
I'm going to close this PR because it's been cherry-picked into the 4.3 branch (or will be shortly)

**Pull request closed without merging**